adapters/repos/db/lsmkv/segment.go

```diff
@@ -127,14 +129,22 @@ func newSegment(path string, logger logrus.FieldLogger, metrics *Metrics,
 	// mmap has some overhead, we can read small files directly to memory
 	var contents []byte
 	var unMapContents bool
-	if size > cfg.MinMMapSize {
+
+	if size <= cfg.MinMMapSize { // check if it is a candidate for full reading
+		err = cfg.allocChecker.CheckAlloc(size) // check if we have enough memory
+		if err != nil {
+			logger.Debugf("memory pressure: cannot fully read segment")
+		}
+	}
+
+	if size > cfg.MinMMapSize || err != nil { // mmap the file if it's too large or if we have memory pressure
```

Comment:
proposition: it's a little bit shadowed that we react to allocChecker error, should we use some bool variable instead? or just rename the error to `allocCheckerErr`?